homeassistant/components/emulated_hue/upnp.py

```diff
@@ -60,13 +68,26 @@ class UPNPResponderThread(threading.Thread):
 
     _interrupted = False
 
-    def __init__(self, host_ip_addr, listen_port, upnp_bind_multicast):
+    def __init__(self, host_ip_addr, listen_port, upnp_bind_multicast,
+                 advertise_ip, advertise_port):
         """Initialize the class."""
         threading.Thread.__init__(self)
 
         self.host_ip_addr = host_ip_addr
         self.listen_port = listen_port
         self.upnp_bind_multicast = upnp_bind_multicast
+        self.advertise_ip = advertise_ip
+        self.advertise_port = advertise_port
+
+        # If using the advertisement overides, only
+        # use them for forming the responses
+        resp_ip = self.advertise_ip if \
```

Comment:
That sounds like a better solution. Good call.

Since this is about some low level stuff, you might want to read some docs about travis's build environment related to the errors.

Also, just as an FYI, you can create an account with travis and have it check pushes to your own fork if you want to try different things out. You're free to push them here as well, but it's nice not to clutter up the PR branch if possible.